FWCore/Framework/src/LuminosityBlockProcessingStatus.h

```diff
@@ -103,10 +105,13 @@ namespace edm {
     std::vector<std::shared_ptr<const EventSetupImpl>> eventSetupImpls_;
     WaitingTaskList endIOVWaitingTasks_;
     edm::WaitingTaskHolder globalEndRunHolder_;
-    std::atomic<unsigned int> nStreamsStillProcessingLumi_{0};  //read/write as streams finish lumi so must be atomic
     edm::Timestamp endTime_{};
-    std::atomic<char> endTimeSetStatus_{0};
+    unsigned int nStreamsProcessingLumi_{0};
```

Comment:
The updates are protected by the spinloops on `state_`, right? How about
```suggestion
    CMS_THREAD_GUARD(state_) unsigned int nStreamsProcessingLumi_{0};
```
?

Reply:
I did this. I assume this is for the static analyzer. I don't understand when this is needed very well. It doesn't hurt anything so I just added it. The variable is only written or read in two non-const functions. Is there a specific reason the static analyzer will complain if this is not added here?

Reply:
In this case, where the variable is modified in non-`const` functions, and therefore is not declared as `mutable`, the static analyzer indeed would not flag it. I'm more thinking of documenting it for human reader (in which case the information could be conveyed also in comments). E.g. it took me a while to realize the updates are protected by the `state_` rather than the `SerialTaskQueue`s.